src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
         except Exception as e:
             raise SystemInfoFetchError(f"Unable to fetch System Info") from e
 
+@dataclass
+class TrainingDetails:
+    hyperparams: dict = None
+    loss_func: str = ""
+    optimizer: str = ""
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["hyperparams"] = self.hyperparams
+        model_card["loss_func"] = self.loss_func
+        model_card["optimizer"] = self.optimizer
+        return model_card
+
+@dataclass
+class EvalDetails:
+    loss_funcs: list = None
+    metrics: list = None
+
+    def toJSON(self) -> Dict:
```

Comment:
the function is called toJSON but we are not returning a JSON string. We return a Dict. 
Do we really need this method? Could you check if calling `json_string = json.dumps(eval_details_object)` would work out of the box?